config/nodeselectors.go

```diff
@@ -0,0 +1,31 @@
+package config
+
+import (
+	"maps"
+	"path"
+)
+
+func getNodeSelectorsByImage(
```

Comment:
could we maybe do a simple unit test w/ a case for matching a glob, no match, and matching default for this one? obv we are just using path.Match here so im not worried about that being broken or anything, but just more wanting to add it from a documentation perspective ya know?